api/pkg/service/auth/auth.go

```diff
@@ -0,0 +1,117 @@
+// Copyright © 2020 The Tekton Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package auth
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/dgrijalva/jwt-go"
+	"github.com/google/go-github/github"
+	"github.com/jinzhu/gorm"
+	"go.uber.org/zap"
+	"golang.org/x/oauth2"
+
+	"github.com/tektoncd/hub/api/gen/auth"
+	"github.com/tektoncd/hub/api/pkg/app"
+	"github.com/tektoncd/hub/api/pkg/db/model"
+)
+
+type service struct {
+	logger        *zap.SugaredLogger
+	db            *gorm.DB
+	conf          *oauth2.Config
+	jwtSigningKey string
+	defaultScopes []string
+}
+
+var (
+	invalidCode   = auth.MakeInvalidCode(fmt.Errorf("invalid authorization code"))
+	internalError = auth.MakeInternalError(fmt.Errorf("failed to authenticate"))
+)
+
+// New returns the auth service implementation.
+func New(api app.Config) auth.Service {
+	return &service{api.Logger(), api.DB(), api.OauthConfig(), api.JWTSigningKey(), api.Data().Default.Scopes}
```

Comment:
we shouldn't Cache the DefaultScopes because the `Config.yaml` can change and the app.Data can get refreshed but the service will still use the old Scopes.

Reply:
Yeah added a `*Data` now.